ansible/roles/kamailio/tasks/main.yml

```diff
@@ -0,0 +1,78 @@
+---
+- name: Install gnupg2
+  become: true
+  apt:
+    name: gnupg2
+    update_cache: true
+
+- name: Add Kamailio GPG apt Key
+  become: true
+  apt_key:
+    url: http://deb.kamailio.org/kamailiodebkey.gpg
+    state: present
+
+- name: Add Kamailio Repository
+  become: true
+  apt_repository:
+    repo: "deb http://deb.kamailio.org/kamailio{{ kamailio_major_version }}{{ kamailio_minor_version }} {{ ansible_distribution_release }} main"
+    state: present
+
+- name: Install packages
+  become: true
+  apt:
+    name:
+      - kamailio
+      - kamailio-tls-modules
+      - kamailio-websocket-modules
+      - kamailio-python3-modules
+      - kamailio-sqlite-modules
+      - default-mysql-server
+      - python3-pip
+    update_cache: true
+
+- name: Install pexpect
+  become: true
+  pip:
+    name: pexpect
+
+- name: Copy configuration files
+  become: true
+  copy:
+    src: "{{ item }}"
+    dest: "/etc/kamailio/{{ item }}"
+  notify: Restart Kamailio
+  loop:
+    - tls.cfg
+    - request_gw.py
+
+- name: Copy configuration templates
+  become: true
+  template:
+    src: "{{ item }}.j2"
+    dest: "/etc/kamailio/{{ item }}"
+  notify: Restart Kamailio
+  loop:
+    - kamailio.cfg
+    - kamctlrc
+    - sipmediagw.cfg
```

Comment:
This file contains secrets and should not be readable by every user: it may be a good idea, if possible, to use the `kamailio` user created when `kamailio` is installed and change ownership modes

(Besides, permissions should always be specified for every `copy`/`template` tasks)